builder/ncloud/step_validate_template.go

```diff
@@ -162,7 +162,7 @@ func (s *StepValidateTemplate) validateMemberServerImage(fnGetServerImageList fu
 
 		if *image.MemberServerImageNo == s.Config.MemberServerImageNo {
 			isExistMemberServerImageNo = true
-			if s.Config.ServerProductCode == "" {
+			if s.Config.ServerProductCode == "" && !s.Config.SupportVPC {
```

Comment:
I'm not really sure how this fixes the issue. Looking at the docs it seems to imply that if SupportVPC is false then the build is running in a classic environment, which means https://github.com/hashicorp/packer-plugin-ncloud/blob/0c050bd243c6debaf564d5a42ab6f5fb1d883ecf/builder/ncloud/step_validate_template.go#L180 would probably return a nil image. Is that correct? 

I feel like there should be a nil check further upstream to validate that image contains all the expected information or is at least not nil. The panic log seems to imply the error happening on line 160 so I'm a little confused here. 

That said, I think it would be nice to see a test case in step_validate_template_test to validate the expected behavior if possible.

Reply:
```GO
step_validate_template.go
49     if config.SupportVPC {
50         step.getZone = step.getZoneCode
51          step.getMemberServerImageList = step.getVpcMemberServerImageList

54     } else {
55         step.getZone = step.getZoneNo
56         step.getMemberServerImageList = step.getClassicMemberServerImageList
57         step.getServerImageProductList = step.getClassicServerImageProductList
58         step.getServerProductList = step.getClassicServerProductList
59     }
60
61     step.Validate = step.validateTemplate
```

if set `support_vpc=true` (VPC)
-> step.getMemberServerImageList = step.getVpcMemberServerImageList
-> func (s *StepValidateTemplate) validateTemplate()
-> func (s *StepValidateTemplate) validateMemberServerImage
  --> fnGetServerImageList()
  --> func (s *StepValidateTemplate) getVpcMemberServerImageList()
         **Here** : vserver.MemberServerImageInstance struct not exist `OriginalServerProductCode`. But, Sets the value not to match. 
**`OriginalServerProductCode` have `r.OriginalServerImageProductCode`value. This is wrong match**
**`OriginalServerImageProductCode` don't have any value(=nil). This is the cause of panic**
```GO
var results []*server.MemberServerImage
	for _, r := range memberServerImageList.MemberServerImageInstanceList {
		results = append(results, &server.MemberServerImage{
			...
                         OriginalServerProductCode:    r.OriginalServerImageProductCode,
		})
	}
```
  --> LINE 169 painc error. Because *image.OriginalServerImageProductCode is nil.
```
169             s.Config.ServerImageProductCode = *image.OriginalServerImageProductCode
```

if set `support_vpc=false` (CLASSIC)
-> step.getMemberServerImageList = step.getClassicMemberServerImageList
-> func (s *StepValidateTemplate) validateTemplate()
-> func (s *StepValidateTemplate) validateMemberServerImage
  --> fnGetServerImageList()
  --> func (s *StepValidateTemplate) getClassicMemberServerImageList()
         **Here** : server.MemberServerImage struct exist `OriginalServerProductCode` and `OriginalServerImageProductCode`.
  --> LINE 169 No painc error. Because both *image.OriginalServerProductCode and *image.OriginalServerImageProductCode values are not nil.